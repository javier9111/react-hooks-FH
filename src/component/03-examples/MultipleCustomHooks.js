import React from "react";
import { useFetch } from "../../Hooks/useFetch";
import "../02-useEffect/effects.css";
export const MultipleCustomHooks = () => {
  const { data, loading, error } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/1`
  );
  // console.log(state);
  const { author, quote } = !!data && data[0];
  return (
    <div>
      <h1>Breaking bad Quotes</h1>
      <hr />
      {loading ? (
        <div className="alert alert-info text-center">loading...</div>
      ) : (
        <blockquote className="blockquote text-right">
          <p className="mb-0">{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}
    </div>
  );
};
