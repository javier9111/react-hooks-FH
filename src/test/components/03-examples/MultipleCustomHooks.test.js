import React from "react";
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../component/03-examples/MultipleCustomHooks";
import { useFetch } from "../../../Hooks/useFetch";
import { useCounter } from "../../../Hooks/useCounter";
jest.mock("../../../Hooks/useFetch"); // here we mock the useFetch hook so everytimen is being used we assign our values
jest.mock("../../../Hooks/useCounter"); // here we mock the useCounter hook so everytimen is being used we assign our values

describe("Test on <MultipleCustomHooks>", () => {
  //since all the test are using this counter hook we put it in a before each block in order to use it, otherwise if get the mockreturn value outside the before each the test will break
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 1,
      increment: () => {},
      decrement: () => {},
    });
  });

  test("should show correctly", () => {
    //here we assign our mock values to be use in this test
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should show the info correctly", () => {
    //here we assign our mock values to be use in this test
    useFetch.mockReturnValue({
      data: [
        {
          author: "John",
          quote: "Hola Mundo",
        },
      ],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-0").text().trim()).toBe("Hola Mundo");
    expect(wrapper.find("footer").text().trim()).toBe("John");
  });
});
