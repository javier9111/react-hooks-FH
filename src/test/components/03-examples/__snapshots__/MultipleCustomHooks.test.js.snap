// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <MultipleCustomHooks> should show correctly 1`] = `
<div>
  <h1>
    Breaking bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    siguiente quote
  </button>
</div>
`;
